src/Graphics/Glyphs/GlyphGeom.cc

```diff
@@ -523,228 +567,632 @@ void GlyphGeom::generateSphere(const Point& center, double radius, int resolutio
-            indices_.push_back(2 + offset);
-            indices_.push_back(1 + offset);
-            indices_.push_back(3 + offset);
-          }
-        for(int jj = 0; jj < 6; jj++) indices_.pop_back();
+        // Rotate norms
+        v1 = rotate * Vector(x1, y1, z1);
+        v2 = rotate * Vector(x2, y2, z2);
       }
+
+      v1.safe_normalize();
+      v2.safe_normalize();
+
+      // Transorm points and add to points list
+      points_.push_back(p1);
+      points_.push_back(p2);
+
+      // Add normals
+      normals_.push_back(v1);
+      normals_.push_back(v2);
+
+      // Add color vectors from parameters
+      colors_.push_back(color);
+      colors_.push_back(color);
+
+      numVBOElements_ += 2;
+
+      indices_.push_back(0 + offset);
+      indices_.push_back(1 + offset);
+      indices_.push_back(2 + offset);
+      indices_.push_back(2 + offset);
+      indices_.push_back(1 + offset);
+      indices_.push_back(3 + offset);
+    }
+    for(int jj = 0; jj < 6; jj++) indices_.pop_back();
+  }
+}
+
+inline double spow(double e, double x)
+{
+  // This for round off of very small numbers.
+  if( abs( e ) < 1.0e-6)
+    e = 0.0;
+
+  if (e < 0.0)
+  {
+    return (double)(pow(abs(e), x) * -1.0);
+  }
+  else
+  {
+    return (double)(pow(e, x));
+  }
+}
+
+void GlyphGeom::generateSuperEllipsoid(const Point& center, Tensor& t, double scale, int resolution, const ColorRGB& color, bool normalize, double emphasis)
+{
+  double zeroThreshold = 0.000001;
+  std::vector<Vector> eigvectors(3);
+  t.get_eigenvectors(eigvectors[0], eigvectors[1], eigvectors[2]);
+
+  double eigval1, eigval2, eigval3;
+  t.get_eigenvalues(eigval1, eigval2, eigval3);
+  Vector eigvals = Vector(fabs(eigval1), fabs(eigval2), fabs(eigval3));
+  if(normalize)
+    eigvals.normalize();
+  eigvals *= scale;
+
+  // Checks to see if eigenvalues are close to 0
+  bool eig_x_0 = eigvals.x() <= zeroThreshold;
+  bool eig_y_0 = eigvals.y() <= zeroThreshold;
+  bool eig_z_0 = eigvals.z() <= zeroThreshold;
+
+  // Set to 0 if below threshold
+  eigvals[0] = (!eig_x_0) * eigvals[0];
+  eigvals[1] = (!eig_y_0) * eigvals[1];
+  eigvals[2] = (!eig_z_0) * eigvals[2];
+
+  bool flatTensor = (eig_x_0 + eig_y_0 + eig_z_0) >= 1;
+  Vector zero_norm;
+
+  if(flatTensor)
+  {
+    reorderTensor(eigvectors, eigvals);
+
+    eig_x_0 = eigvals.x() <= zeroThreshold;
+    eig_y_0 = eigvals.y() <= zeroThreshold;
+    eig_z_0 = eigvals.z() <= zeroThreshold;
+    // Check for zero eigenvectors
+    if(eig_x_0)
+    {
+      zero_norm = Cross(eigvectors[1], eigvectors[2]);
+      eigvectors[0] = zero_norm;
+    }
+    else if(eig_y_0)
+    {
+      zero_norm = Cross(eigvectors[0], eigvectors[2]);
+      eigvectors[1] = zero_norm;
+    }
+    else if(eig_z_0)
+    {
+      zero_norm = Cross(eigvectors[0], eigvectors[1]);
+      eigvectors[2] = zero_norm;
+    }
+  }
+
+  Transform rotate(Point(0.0, 0.0, 0.0), eigvectors[0], eigvectors[1], eigvectors[2]);
```

Comment:
This is duplicated above--make a new function `transformFromEigenVectors` or something